### ISSUE TITLE
Events  show up  only once(via webhooks work with dingding robot) under an alert , which are after being registered.that means dingding robot reveive message only once,which is not right,it should be saw many times

### ISSUE THREAD

**Issue by nicky132** [getsentry/self-hosted#2423]
### Problem Statement

一、
1、It says i have given many errors in my project
2、sentry also triggered many times,which are correct
![image](https://github.com/getsentry/self-hosted/assets/34881461/caba5f2a-bf46-4314-b4e1-92cb1f5ea197)
![image](https://github.com/getsentry/self-hosted/assets/34881461/2c64ba62-687e-44fc-8961-a43aab0ef636)

二、rules are correct,are they?
![image](https://github.com/getsentry/self-hosted/assets/34881461/79a91f54-c85d-4418-bd61-670a8c14700a)

三、why i got warning message with dingding robot via webhooks only once,only once,it doesn‘t right.
![image](https://github.com/getsentry/self-hosted/assets/34881461/9fb1f60b-f814-4e23-9f59-3f4b8da949b5)

### Solution Brainstorm

Maybe something wrong with kafka?

Information i can off here:
sentry:[23.8.0](https://github.com/getsentry/self-hosted/releases/tag/23.8.0)
docker:22.0.5
os:centos7
linux:x86_64

I hope i can got warn message correctly every time ,which sentry is triggered by application occurs problem

**Comment by hubertdeng123** [comment]
Are you receiving an alert once for every type of error? From the rules of your alert, it seems like the alert would only fire if a new type of issue occurs, not when an old issue is thrown again. It is normal to have only one alert for a type of issue, even though an error may be triggered multiple times.

**Comment by nicky132** [comment]
> Are you receiving an alert once for every type of error? From the rules of your alert, it seems like the alert would only fire if a new type of issue occurs, not when an old issue is thrown again. It is normal to have only one alert for a type of issue, even though an error may be triggered multiple times.

Yes,i do receive an alert once for every type of error.but if i want every issue to trigger alert,no matter it was new issue or old issue.what can i do?
Please let me know, if you know that.

By the way,why i got so many times "504 Gateway Time-out" when i access my sentry.

**Comment by nicky132** [comment]
> > Are you receiving an alert once for every type of error? From the rules of your alert, it seems like the alert would only fire if a new type of issue occurs, not when an old issue is thrown again. It is normal to have only one alert for a type of issue, even though an error may be triggered multiple times.
> 
> Yes,i do receive an alert once for every type of error.but if i want every issue to trigger alert,no matter it was new issue or old issue.what can i do? Please let me know, if you know that.
> 
> By the way,why i got so many times "504 Gateway Time-out" when i access my sentry.

![1695781152054_DF680EB0-49AC-4cac-89E5-63BBCDF332C2](https://github.com/getsentry/self-hosted/assets/34881461/0d866695-8272-4e09-af9e-ca6accbc4c13)

**Comment by hubertdeng123** [comment]
> Yes,i do receive an alert once for every type of error.but if i want every issue to trigger alert,no matter it was new issue or old issue.what can i do? Please let me know, if you know that.

It is working as intended then. If you'd like to be notified off every issue, you can try the alert rule involving number of errors triggered in a certain period of time.

> By the way,why i got so many times "504 Gateway Time-out" when i access my sentry.

You may want to check your nginx logs to figure that out

**Comment by nicky132** [comment]
> > Yes,i do receive an alert once for every type of error.but if i want every issue to trigger alert,no matter it was new issue or old issue.what can i do? Please let me know, if you know that.
> 
> It is working as intended then. If you'd like to be notified off every issue, you can try the alert rule involving number of errors triggered in a certain period of time.
> 
> > By the way,why i got so many times "504 Gateway Time-out" when i access my sentry.
> 
> You may want to check your nginx logs to figure that out

<img width="1071" alt="image" src="https://github.com/getsentry/self-hosted/assets/34881461/720db3b1-5132-44e4-a7ec-f64dc06976d8">
<img width="1084" alt="image" src="https://github.com/getsentry/self-hosted/assets/34881461/a5d64d66-5eb1-45f9-8042-87a5453f26fe">
Thank you, but just let me know,which one should i choose

**Comment by azaslavsky** [comment]
Alerts work on a per-issue, not per-event, basis. If you mark an issue as ignored or resolved and it re-occurs, you should get an alert. But merely receiving more events is not enough to trigger one, and would result in a flood of noisy alerts that drown out all signal in most cases.

**Comment by nicky132** [comment]
> Alerts work on a per-issue, not per-event, basis. If you mark an issue as ignored or resolved and it re-occurs, you should get an alert. But merely receiving more events is not enough to trigger one, and would result in a flood of noisy alerts that drown out all signal in most cases.

![image](https://github.com/getsentry/self-hosted/assets/34881461/166a19ca-2244-4f31-8e78-8b9d8180cf7c)
I totaly agree with you ,Alerts work on a per-issue,but i did get an new and same issue, then, sentry webhooks did not callback my Application Programming Interface, but i do want every issue should be warned by robot via webhooks.
What should i do?

**Comment by azaslavsky** [comment]
I'm not sure I understand the problem you are describing. Do you get any pings to your webhook via the API app at all? Or is it pinging for new issues, but not recurrences of existing issues?

**Comment by nicky132** [comment]
> I'm not sure I understand the problem you are describing. Do you get any pings to your webhook via the API app at all? Or is it pinging for new issues, but not recurrences of existing issues?

I did get ping only when new issue  occurred, but not for existing issues.
That means if you recurrence  the same issue, you never get any pings.

**Comment by azaslavsky** [comment]
Unless the issue was previously marked as resolved, I believe this is working as intended?

**Comment by nicky132** [comment]
> Unless the issue was previously marked as resolved, I believe this is working as intended?

You can say that, though.
But if i want to let robot tell me every warning error ,no matter its new issue or the same existing issue, every single issue i need to know by robot via webhook.

**Comment by azaslavsky** [comment]
> But what if i want to let robot tell me every warning error ,no matter its new issue or the same existing issue, every single issue i need to know by robot via webhook.

This isn't a possibility with the current system - it is not supported. I'm sorry!
